File: ios/Classes/FlutterContactIndexPlugin.swift
import Flutter
import UIKit

public class FlutterContactIndexPlugin: NSObject, FlutterPlugin {
    
    private static var api: FlutterContactIndexHostImpl? = nil
    public static func register(with registrar: FlutterPluginRegistrar) {
        api = FlutterContactIndexHostImpl()
        FlutterContactIndexHostApiSetup.setUp(
            binaryMessenger: registrar.messenger(),
            api: api
        )
    }
    
    public static func unregister(with registrar: FlutterPluginRegistrar) {
        // 注销 FlutterContactIndexHostApiSetup 实例，传入 nil 会清除消息处理程序
        FlutterContactIndexHostApiSetup.setUp(
            binaryMessenger: registrar.messenger(),
            api: nil
        )
        api = nil
    }
}

File: android/src/main/kotlin/com/hzm/flutter_contact_index/FlutterContactIndexPlugin.kt
package com.hzm.flutter_contact_index

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** FlutterContactIndexPlugin */
class FlutterContactIndexPlugin : FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private var channel: FlutterContactIndexHostImpl? = null

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = FlutterContactIndexHostImpl(flutterPluginBinding.applicationContext)
    FlutterContactIndexHostApi.setUp(flutterPluginBinding.binaryMessenger, channel)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {

  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    FlutterContactIndexHostApi.setUp(binding.binaryMessenger, null)
    channel?.cleanup()
    channel = null
  }
}
